ops/internal/deployer/deployer.go

```diff
@@ -0,0 +1,308 @@
+	if err := cmd.Run(); err != nil {
+		return nil, fmt.Errorf("failed to run op-deployer apply: %w", err)
+	}
+
+	// Run `op-deployer inspect genesis` to read the expected genesis
+	d.lgr.Info("Running `op-deployer inspect genesis`")
+	cmd = exec.Command(d.binaryPath, "inspect", "genesis", "--workdir", workdir, chainId)
+
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to run op-deployer inspect genesis: %w", err)
+	}
+
+	var genesis OpaqueMap
+	if err := json.Unmarshal(output, &genesis); err != nil {
+		return nil, fmt.Errorf("failed to parse op-deployer inspect genesis output: %w", err)
+	}
+
+	return &genesis, nil
+}
+
+func (d *OpDeployer) copyStateFileToTempDir(statePath string) (string, error) {
+	// Create a temporary directory
+	workdir, err := os.MkdirTemp("", "op-deployer")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temporary directory: %w", err)
+	}
+
+	state, err := ReadOpaqueStateFile(statePath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read state file: %w", err)
+	}
+
+	// Write state.json in the temp directory
+	stateJSON, err := json.MarshalIndent(state, "", "  ")
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal state to JSON: %w", err)
+	}
+
+	stateTempPath := filepath.Join(workdir, "state.json")
+	if err := os.WriteFile(stateTempPath, stateJSON, 0o644); err != nil {
+		return "", fmt.Errorf("failed to write state to temp file: %w", err)
+	}
+
+	d.lgr.Info("Copied state file to temporary directory", "path", stateTempPath)
+	return workdir, nil
+}
+
+func (d *OpDeployer) InspectGenesis(statePath, chainId string) (*OpaqueMap, error) {
+	// Run `op-deployer inspect genesis` to read the expected genesis
+	d.lgr.Info("Running `op-deployer inspect genesis`")
+
+	workdir, err := d.copyStateFileToTempDir(statePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to copy state file: %w", err)
+	}
+	defer os.RemoveAll(workdir)
+
+	cmd := exec.Command(d.binaryPath, "inspect", "genesis", "--workdir", workdir, chainId)
+	stderr := bytes.Buffer{}
+	cmd.Stderr = &stderr
+	output, err := cmd.Output()
+	if err != nil {
+		d.lgr.Error(stderr.ReadString(0))
+		return nil, fmt.Errorf("failed to run op-deployer inspect genesis: %w", err)
+	}
+
+	var genesis OpaqueMap
+	if err := json.Unmarshal(output, &genesis); err != nil {
+		return nil, fmt.Errorf("failed to parse op-deployer inspect genesis output: %w", err)
+	}
+
+	return &genesis, nil
+}
+
+func (d *OpDeployer) InspectRollup(statePath, chainId string) (*rollup.Config, error) {
+	// Run `op-deployer inspect rollup` to read the expected rollup config
+	d.lgr.Info("Running `op-deployer inspect rollup`")
+
+	workdir, err := d.copyStateFileToTempDir(statePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to copy state file to temporary directory: %w", err)
+	}
+	defer os.RemoveAll(workdir)
+
+	cmd := exec.Command(d.binaryPath, "inspect", "rollup", "--workdir", workdir, chainId)
+	stderr := bytes.Buffer{}
```

Comment:
[nit] for these buffers it's safer to do new(bytes.Buffer), since it's bad practice to copy the buffer by value

Reply:
Done in https://github.com/ethereum-optimism/superchain-registry/pull/1041/commits/1fba7e052bf9101c629c66b62fe5ee4e66b0c02d